### ISSUE TITLE
Adding label smoothing option to `nn.BCELoss`  and `nn.BCEWithLogitsLoss`?

### ISSUE THREAD

**Issue by dyhan316** [pytorch/pytorch#91545]
### 🚀 The feature, motivation and pitch

Hello, I was looking at BCELoss and CrossEntropyLoss and found that while the latter has the option to add label_smoothing, the former does not. 

### Alternatives

Seeing as how the reasoning used for performing label smoothing (i.e. the logit values having to go to $\pm\infty$ to approach 1 or 0) should also apply to sigmoid functions (which are used with BCELoss), it stands to reason that label smoothing can and should be implemented with BCELoss? (i.e. for ex :  if the label value is 1, with label smoothing of 0.2, the value should be 0.8?)

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by jbschlosser** [comment]
[Keras has it](https://www.tensorflow.org/api_docs/python/tf/keras/losses/BinaryCrossentropy), so I think we should too. We'd accept a PR implementing it if you'd like to contribute.

**Comment by dyhan316** [comment]
@albanD  Thank you for your reply! I meant having the same label smoothing option as in `nn.CrosEntropyLoss` for  `nn.BCELoss` and `nn.BCEWithLogitsLoss`.

**Comment by albanD** [comment]
Ok!
@jbschlosser what about a python side implementation of this before we call into the c++ backend?

**Comment by dyhan316** [comment]
Thank you for your reply! Also, I think that one thing that might have to be considered is, `BCELoss` only takes in float tensors as targets (https://github.com/pytorch/pytorch/issues/2220) therefore, if for some reason the target values are neither exactly 0.0 or 1.0, simply applying `label_smoothing` by addition/subtraction might be weird?

**Comment by jbschlosser** [comment]
> Ok! @jbschlosser what about a python side implementation of this before we call into the c++ backend?

I like this; makes it much easier to introduce without dealing with BC issues, etc.

> if for some reason the target values are neither exactly 0.0 nor 1.0, simply applying label_smoothing by addition/subtraction might not make sense?

Note that it's already possible to use `CrossEntropyLoss` with float labels and `label_smoothing`. We assume the user knows what they're doing mixing these two. Maybe they're trying out some advanced form of label smoothing?

**Comment by dyhan316** [comment]
Oh i see! Better to let the user decide for themselves rather than restricting it.

**Comment by jbschlosser** [comment]
@dyhan316 We'd accept a PR implementing this on the python side if you'd like to contribute!

**Comment by 00itamarts00** [comment]
Hi, 
I would like to implement this feature. Is that Ok?

**Comment by apple2373** [comment]
```
class BCEWithLogitsLoss(nn.Module):
    def __init__(self, label_smoothing=0.0, reduction='mean'):
        super(BCEWithLogitsLoss, self).__init__()
        assert 0 <= label_smoothing < 1, "label_smoothing value must be between 0 and 1."
        self.label_smoothing = label_smoothing
        self.reduction = reduction
        self.bce_with_logits = nn.BCEWithLogitsLoss(reduction=reduction)

    def forward(self, input, target):
        if self.label_smoothing > 0:
            positive_smoothed_labels = 1.0 - self.label_smoothing
            negative_smoothed_labels = self.label_smoothing
            target = target * positive_smoothed_labels + \
                (1 - target) * negative_smoothed_labels

        loss = self.bce_with_logits(input, target)
        return loss
```

How about this?

**Comment by waylonflinn** [comment]
Looks like the PR for this issue got reverted at some point (Aug 13, 2025) and it hasn't actually been completed. Should this be reopened? This feature would be useful for me.

Relevant PR: https://github.com/pytorch/pytorch/pull/150282
Revert: https://github.com/pytorch/pytorch/pull/150282#issuecomment-3182854674
